bin/echo.c

```diff
@@ -376,23 +376,25 @@ int echo(struct s2n_connection *conn, int sockfd, bool *stop_echo)
 
             if (readers[0].revents & POLLIN) {
                 s2n_errno = S2N_ERR_T_OK;
-                bytes_read = s2n_recv(conn, buffer, STDIO_BUFSIZE, &blocked);
-                if (bytes_read == 0) {
-                    return 0;
-                }
-                if (bytes_read < 0) {
-                    switch (s2n_error_get_type(s2n_errno)) {
-                        case S2N_ERR_T_BLOCKED:
-                            /* Wait until poll tells us data is ready */
-                            continue;
-                        case S2N_ERR_T_ALERT:
-                            fprintf(stderr, "Received alert: %d\n", s2n_connection_get_alert(conn));
-                            break;
-                        default:
-                            fprintf(stderr, "Error reading from connection: '%s'\n", s2n_strerror(s2n_errno, "EN"));
-                            break;
+                while (1) {
+                    bytes_read = s2n_recv(conn, buffer, STDIO_BUFSIZE, &blocked);
+                    if (bytes_read == 0) {
+                        return 0;
+                    }
+                    if (bytes_read < 0) {
+                        switch (s2n_error_get_type(s2n_errno)) {
+                            case S2N_ERR_T_BLOCKED:
+                                /* Wait until poll tells us data is ready */
+                                break;
```

Comment:
Don't we need to loop on poll again in the case where s2n_recv returns blocked? Currently it looks like this will break out of the switch statement and hit the exit().

Reply:
doh, indeed, looks like I copied this change incorrectly - I need to remove that exit or move it to the other 2 branches.